ddtrace/tracer/example_test.go

```diff
@@ -3,37 +3,47 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016 Datadog, Inc.
 
-package tracer
+package tracer_test
 
 import (
-	"io/ioutil"
+	"context"
 	"log"
+	"math/rand"
 
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 )
 
 // A basic example demonstrating how to start the tracer, as well as how
 // to create a root span and a child span that is a descendant of it.
 func Example() {
+	ctx := context.Background()
+
 	// Start the tracer and defer the Stop method.
-	Start(WithAgentAddr("host:port"))
-	defer Stop()
+	tracer.Start()
+	defer tracer.Stop()
 
 	// Start a root span.
-	span := StartSpan("get.data")
+	span, ctx := tracer.StartSpanFromContext(ctx, "your.work")
 	defer span.Finish()
 
-	// Create a child of it, computing the time needed to read a file.
-	child := StartSpan("read.file", ChildOf(span.Context()))
-	child.SetTag(ext.ResourceName, "test.json")
+	yourCode := func(ctx context.Context) error {
```

Comment:
Why does `yourCode` need to be an inline function? I'm not sure if it's actually creating a good conceptual separation or simply introducing more confusion. I would suggest either:
* Moving into a separate function from `main()`. I think it's OK if it doesn't show up in the example on GoDoc. It will still be runnable. In that case maybe call it `PerformOperation(ctx)` to make it self-documented by name.
* Inline the code like before.

WDYT?

Reply:
The example will still be self-contained and runnable. I'm not sure how it shows up in GoDoc though if you turn `yourCode` into a separate function. I have not seen this pattern before and I want to make sure the example is focused on the point without the noise. I would try to put `yourCode` in a separate function outside `main`. The example will be runnable, but not sure how godoc would show it. 

I'll leave it up to you.

Reply:
Moved it out. For some reason I thought that if I moved it out of the `Example` function then it wouldn't show up in pkg.go.dev. But according to the docs, it should be fine:

> The entire test file is presented as the example when it contains a single example function, at least one other function, type, variable, or constant declaration, and no test or benchmark functions.